app/jobs/nudge_users_to_request_mentoring_job.rb

```diff
@@ -0,0 +1,43 @@
+class NudgeUsersToRequestMentoringJob < ApplicationJob
+  queue_as :notifications
+
+  def perform
+    # Don't change this without actively benchmarking it on bastion.
+    # Things that logically feel like they should make it faster often don't.
+    # At time of writing it takes 4.8s.
+    users = User.where(
+      id: Iteration.joins(solution: :exercise).
+        # Iterated since v3 launch
+        where('iterations.created_at >= ?', Date.new(2021, 9, 1)).
+        # Iterated over a day ago (give time to request manually)
+        where('iterations.created_at < ?', Time.current - 1.day).
+
+        # Don't include Concept Exercises as we don't nudge on those
+        where('solutions.type': 'PracticeSolution').
+
+        # Don't include hello-world
+        where.not('exercises.slug': 'hello-world').
+
+        # Not for users who have mentor requests
+        where.not('solutions.user_id': Mentor::Request.select(:student_id)).
+
+        # Not for users who have already had the notification
+        where.not('solutions.user_id': User::Notifications::NudgeToRequestMentoringNotification.select(:user_id)).
+
+        # Just the user ids
+        select('solutions.user_id')
+    )
+
+    users.find_each do |user|
+      track = PracticeSolution.status_iterated.where(user: user).
+        joins(:exercise).where.not('exercises.slug': 'hello-world').
+        last.track
```

Comment:
Does `.last` by default return the latest track? I'm asking because there is no explicit sorting here.

Reply:
It will return the last practice solution, which will be default ordered by the primary key, so it'll be the last submitted.